src/sgame/sg_bot_nav.cpp

```diff
@@ -546,6 +546,8 @@ bool BotAvoidObstacles( gentity_t *self, vec3_t dir )
 
 	if ( blocker )
 	{
+
+		self->botMind->cmdBuffer.upmove = -127;
```

Comment:
I don't understand. When would this ever be a good idea?

Reply:
no and for 2 reasons:

* it's a bad idea to do that unconditionally, but this is something which required much work
* the most important reason, and the why I wanted your input: I remember reading code saying it was a bad idea as it can break prediction, or something like that. Also, the value is clearly something I have copied without understanding it.

I have since fixed the 1st reason, but I still don't like inserting a "random" value in there.

Reply:
Er, that was not a yes-or-no question... why would it be a good idea to "duck when stuck"? What problem does that solve?

Reply:
When the navmesh is generated for areas in which bots needs crouching (this require a minor patch in daemonmap to use the crouched mins/maxs), the bots will get stucked/blocked when trying to follow the path.
Ducking in those conditions (and only those, but I can imagine ducking when meeting a friend so that friend might jump over, too) is the only way to allow reaching vents and the like (lot of those in parpax, for example).

Reply:
> When the navmesh is generated for areas in which bots needs crouching (this require a minor patch in daemonmap to use the crouched mins/maxs), the bots will get stucked/blocked when trying to follow the path.

You really need put this kind of thing in the code comments and commit message.

Have you tested this and the modified daemonmap together successfully?

Maybe a trace with the box from crouchMaxs to maxs could be done, instead of mins to maxs, to test for this situation? Then crouching would not wrongly be triggered outside the vents.

Reply:
I have tested, and am right now in the process of writing code which checks (with trace) if the ducking is necessary (the code works, but I think it can be optimised, which is what I am doing).

You can see the results (of my current code, not the one here, I improved on the detection, but also need bigger live verifications) here: http://deadbeef.fr/random/crouch_and_climb.ogg